### ISSUE TITLE
Message Interfaces for Inter-Module Communication

### ISSUE THREAD

**Issue by skugele** [CognitiveComputingResearchGroup/lidapy-framework#4]
I've stubbed out a set of "message classes" in both the Python LIDA framework and ROS for communications between the LIDA modules.  We need to determine what these messages should look like and whether the current message classes are sufficient / desirable.

This is the list so far:

- Behavior
- Coalition
- ConsciousContent
- Cue
- Episode
- Feature
- Percept
- SpatialMap

One possible problem is that these messages are all "singular" nouns, but we may need to make them plural (e.g., Behaviors rather than Behavior).

I'm creating this issue to help organize the discussion.

@CognitiveComputingResearchGroup/python-lida-framework

**Comment by pulinagrawal** [comment]
The messages to and from the Environment module?
Sensory Stimulus and Actions

Are the implicit because they are handled by ROS?

**Comment by sfstrain** [comment]
Guys--I'll be at CCRG tomorrow afternoon, but other than that, I won't be available for meetings tomorrow. Cheers, Steve 

    On Friday, May 6, 2016 4:07 PM, Pulin Agrawal <notifications@github.com> wrote:
 
 

 The messages to and from the Environment module?
Sensory Stimulus and ActionsAre the implicit because they are handled by ROS?—

**Comment by skugele** [comment]
Closing this topic as it has not had activity in several months, and I found an alternate approach that was more flexible.

SUMMARY: I made the decision to make all messages of type std_msgs:string.  Python classes are "auto-magically" serialized/deserialized by the python framework behind the scenes using cpickle.  This allows any python class to serve as the message type, and does not require any custom messages.
